tests/cuda_test/test_kernel_semantic.py

```diff
@@ -121,4 +121,103 @@ def kernel_call():
     # Check that the error is correct
     error_info = [*errors.error_info_map.values()][0][0]
     assert error_info.symbol.func  == 'kernel_call'
-    assert UNVALID_KERNEL_CALL_TP_BLOCK == error_info.message
+    assert INVALID_KERNEL_CALL_TP_BLOCK == error_info.message
+
+@pytest.mark.parametrize( 'language', [
+        pytest.param("ccuda", marks = pytest.mark.ccuda)
+    ]
+)
+def test_invalid_kernel_config(language):
+    def invalid_kernel_config():
+        @kernel
+        def kernel_call():
+            pass
+        kernel_call[1.0, 2.0]()
+    # Initialize singleton that stores Pyccel errors
+    errors = Errors()
+
+    # epyccel should raise an Exception
+    with pytest.raises(PyccelSemanticError):
+        epyccel(invalid_kernel_config, language=language)
+
+    # Check that we got exactly 2 Pyccel error
+    assert errors.has_errors()
+    assert errors.num_messages() == 2
+
+    # Check that the errors are correct
+    error_info = [*errors.error_info_map.values()][0]
+    assert error_info[0].symbol.func == 'kernel_call'
+    assert error_info[1].symbol.func == 'kernel_call'
+    assert INVALID_KERNEL_CALL_BP_GRID == error_info[0].message
+    assert INVALID_KERNEL_CALL_TP_BLOCK == error_info[1].message
+
+@pytest.mark.parametrize( 'language', [
+        pytest.param("ccuda", marks = pytest.mark.ccuda)
+    ]
+)
+def test_missing_kernel_config(language):
+    def missing_kernel_config():
+        @kernel
+        def kernel_call():
+            pass
+        kernel_call()
+
+    errors = Errors()
+
+    with pytest.raises(PyccelSemanticError):
+        epyccel(missing_kernel_config, language=language)
+
+    assert errors.has_errors()
+    assert errors.num_messages() == 1
+
+    error_info = [*errors.error_info_map.values()][0][0]
+    assert error_info.symbol.func_name == 'kernel_call'
+    assert MISSING_KERNEL_CONFIGURATION == error_info.message
+
+@pytest.mark.parametrize( 'language', [
+        pytest.param("ccuda", marks = pytest.mark.ccuda)
+    ]
+)
+def test_invalid_block_number(language):
+    def invalid_block_number():
+        @kernel
+        def kernel_call():
+            pass
+        blocks_per_grid = 5.0
+        kernel_call[blocks_per_grid, 1]()
+
+    errors = Errors()
+
+    with pytest.raises(PyccelSemanticError):
+        epyccel(invalid_block_number, language=language)
+
+    assert errors.has_errors()
+    assert errors.num_messages() == 1
+
+    error_info = [*errors.error_info_map.values()][0][0]
+    assert error_info.symbol.func == 'kernel_call'
+    assert  INVALID_KERNEL_CALL_BP_GRID == error_info.message
+
+@pytest.mark.parametrize( 'language', [
+        pytest.param("ccuda", marks = pytest.mark.ccuda)
+    ]
+)
+def test_invalid_thread_per_block(language):
+    def invalid_thread_per_block():
+        @kernel
+        def kernel_call():
+            pass
+        threads_per_block = 5.0
+        kernel_call[1, threads_per_block]()
```

Comment:
Why is the number of threads a float?

Reply:
The same here, I provided invalid type to raise the appropriate error.